skopt/optimizer/optimizer.py

```diff
@@ -172,7 +173,10 @@ def __init__(self, dimensions, base_estimator="gp",
                           DeprecationWarning)
             n_initial_points = n_random_starts
 
-        if isinstance(base_estimator, str) and base_estimator.upper() == "GP":
+        self._check_arguments(base_estimator, n_initial_points, acq_optimizer,
+                              dimensions)
+
+        if isinstance(self.base_estimator_, GaussianProcessRegressor):
```

Comment:
switched the order of `_check_arguments` and this test so that we can handle `"GP"` and the user passing a GPRegressor instance